### ISSUE TITLE
Install/set-up Google Site Kit

### ISSUE THREAD

**Issue by lukefretwell** [proudcity/wp-proudcity#1472]
- [x] Test/configure internally
- [x] Launch platform-wide

* Official plugin: https://sitekit.withgoogle.com/
* Details: https://sitekit.withgoogle.com/news/introducing-site-kit-by-google-for-wordpress/
* Call for beta testers: https://docs.google.com/forms/d/e/1FAIpQLScuPDz9VlslHwFfeV-PrmWJoRy6ia9eXWtoar2fTHyMJEEwbQ/viewform

From Google:

Managing a successful website can be complicated. Publishers and business owners should have easy access to the insights and capabilities that drive audience and revenue growth. That’s why we’re excited to announce Site Kit by Google, a free, open-source plugin bringing together insights from essential Google products for WordPress sites.

Site Kit is ideal for website owners who want easily digestible information about audience and revenue performance, all in one place.

In the first version of the plugin, we’ve integrated four Google tools and products to help grow, optimize and monetize WordPress sites:

Search Console: Learn how users discover your content on Google Search.
Analytics: Understand how users navigate your site.
AdSense: Set up AdSense and monitor your earnings.
PageSpeed Insights: Identify critical performance optimizations for your site.
With Site Kit installed, WordPress users can access unified insights and Google product capabilities directly from the WordPress admin panel. Where it is helpful, Site Kit will also provide deep links into Google products for advanced reports and product configuration capabilities.

Site Kit will open to beta testers in early 2019. We’d love to hear what you think! If you’re willing to provide early feedback to help us build a great experience for WordPress, please request early access and product updates.

**Comment by lukefretwell** [comment]
Notes:

* Plugin: https://wordpress.org/plugins/google-site-kit/
* Dashboard: https://sitekit.withgoogle.com/documentation/using-site-kit/using-the-site-kit-dashboard/
* Blog: https://sitekit.withgoogle.com/news/
* Documentation: https://sitekit.withgoogle.com/documentation/
* Settings: https://sitekit.withgoogle.com/documentation/using-site-kit/settings/
* Limitations info: https://sitekit.withgoogle.com/documentation/using-site-kit/known-limitations/

**Comment by kevindherman** [comment]
@lukefretwell @curtismchale are these items at the end of the limitations a blocker:
WordPress setups
- Multisite: Currently, Site Kit doesn’t support [WordPress multisite networks](https://wordpress.org/support/article/create-a-network/). If you want to use Site Kit in a WordPress multisite installation, activate the plugin for each individual site you would like to use it on.
- Headless: Headless setups are not supported at this time.

**Comment by lukefretwell** [comment]
@kevindherman I don't think so but @curtismchale would know better.

**Comment by kevindherman** [comment]
@curtismchale thank you for confirming.

In my quick look at the limitations those were the only questions I had. Seems like it has a lot of potential.

**Comment by lukefretwell** [comment]
The install looks great. 

Questions:

* Seems like there's one main Google account attached to this. How should we handle this? A ProudCity account or an internal user?

Comments:

* I feel comfortable going this route for the analytics upgrade if we can add events and items @kevindherman mentioned. @curtismchale sounds like you said that's doable, but I didn't quite get how we'd implement.

Notes if we dev:

* SiteKit dropdown on live site has spacing between the admin navbar and dropdown.
* When SiteKit is selected in the admin, the icon in the leftnav is white, making it non-accessible.

**Comment by curtismchale** [comment]
@lukefretwell @kevindherman 

I'd say one single PC account `dev@` or something like that.

It should be doable. Currently we have a JavaScript file in our the [ProudCity Dashboard plugin](https://github.com/proudcity/proudcity-dashboard/blob/master/app/js/routes/analyticsRoutes.js) that sends different events to Google Analytics. We'd need to duplicate this functionality for GA4. I'd vote for dropping the current code entirely and building something new as it's in AngularJS and I'd like to move away  from that.

I think your final two points about the spacing and white icon should be able to be resolved fairly easily with some CSS.

For the icon we'll need to use a [CSS filter to get the colour we want](https://stackoverflow.com/questions/22252472/how-can-i-change-the-color-of-an-svg-element).

For the spacing, some negative `margin-top` does the trick, though there is probably a better way than that once I really dig into it.

**Comment by lukefretwell** [comment]
I think we're set on moving away from angular. Let's discuss on our Tuesday call whether SiteKit is the best approach and how we'd deploy:

* additional information re: events
* what account to use for setting it up

**Comment by lukefretwell** [comment]
@curtismchale looks like the white space issue for the drop-down on the front-end isn't a problem anymore.

**Comment by lukefretwell** [comment]
Seeing this so note for how we connect to a primary account. This reads like we should set up each site using a unique account owned by the site owner.

https://sitekit.withgoogle.com/documentation/getting-started/install/#additional-services

```
Managing multiple sites with Site Kit

Site Kit requires connection to a Google account. Note that at this time, you are limited to a total of 100 site connections per Google account. This is an [OAuth 2.0 limitation,](https://developers.google.com/identity/protocols/oauth2#:~:text=There%20is%20currently%20a%20limit,not%20apply%20to%20service%20accounts.) the authentication tool used to verify users’ Google accounts.

You can view all connected sites by going to Site Kit > Dashboard, clicking on your email address, and selecting Manage Sites.
The Manage sites link on the Site Kit dashboard

When you connect a 101st site, the first site that you connected will be disconnected , so we recommend regularly reviewing and removing site connections that are no longer needed.
A list of Site Kit connected sites

Only one Google account per WordPress user can be connected to Site Kit. For example, if you complete the initial Site Kit setup with the Google account user1@gmail.com and later connect AdSense with user2@gmail.com, Site Kit will break the connection with user1@gmail.com and only retain the connection with user2@gmail.com. If user2@gmail.com does not have access to the same services as user1@gmail.com, you may see permissions errors on the Site Kit dashboard.

To ensure that you can successfully set up and access all of your desired Google services via Site Kit, make sure that you have a single Google account that has full access permissions for all of the appropriate Google services
```

**Comment by lukefretwell** [comment]
Got these errors while trying to connect to more services > tag manager.

![Screenshot 2023-03-17 at 15-52-23 Site Kit by Google Dashboard ‹ ProudCity — WordPress](https://user-images.githubusercontent.com/1569463/226066227-dd54da87-9731-493e-a98c-1c70a135225e.png)
![Screenshot 2023-03-17 at 15-51-34 Site Kit by Google Dashboard ‹ ProudCity — WordPress](https://user-images.githubusercontent.com/1569463/226066232-74f86afc-a714-4814-8c87-fa3d3b5600a5.png)

And now can't see tag manager option:

![Screenshot 2023-03-17 at 15-56-13 Site Kit by Google Settings ‹ ProudCity — WordPress](https://user-images.githubusercontent.com/1569463/226066273-9efbd4ca-f4e1-4c22-89ba-023f282c88b5.png)

**Comment by curtismchale** [comment]
Note that every user that wants to see Site Kit Analytics will need to have a Google Account that has access to the Analytics for the site.

How does this work/look for you @kevindherman @lukefretwell? At this point I've had to add `manage_options` to the capabilities of our editors (support on our site) which gives far too many permissions because Google Mapped their capabilities onto `manage_options`. Still seeing if I can unroll that.

<img width="1370" alt="CleanShot 2023-03-21 at 13 38 54@2x" src="https://user-images.githubusercontent.com/74561/226735205-6ed6f98d-83b9-422d-95f4-c5468340c162.png">

**Comment by lukefretwell** [comment]
@kevindherman @curtismchale here are some dashboard options:

* [SB Admin 2](https://startbootstrap.com/theme/sb-admin-2) (free)
* [Boostrap Themes](https://themes.getbootstrap.com/shop/?s=dashboard) (paid)

I think it's worth spending the money to get a theme especially if it has design components for other dashboard related pages.

For BS Themes [Phoenix](https://themes.getbootstrap.com/product/phoenix-admin-dashboard-webapp-template/) looks good to me. Looks like we'd have to pay $139 for multisite. [What we get](https://themes.getbootstrap.com/preview/?theme_id=120099) with that.

Considerations:

* Clean aesthetic (no embellishments)
* Easy to customize
* Flexible to use with other areas of the dashboard

**Comment by curtismchale** [comment]
@lukefretwell I read through the licensing details and I think we need the Extended version as we have a paid product of which this is a feature: https://themes.getbootstrap.com/licenses/

**Comment by lukefretwell** [comment]
- [ ] Move 'Analytics' tab directly below 'View site'
- [ ] 'Analytics' icon is showing big on non 'Analytics' admin pages and Google Site Kit page.
- [ ] Possible to use `<sup>` for BETA? 
- [ ] Use `#0071bc` for buttons and link colors in 'Analytics' pages (Help, Site Kit content section).
- [ ] Text color should be our default black. It's a little tough to read the dark gray.
- [ ] 'Analytics' landing page should probably just have the cards that link to 'Insights', 'Site Kit' and 'Help'.
- [ ] Remove/hide 'Site Kit' link in main leftnav admin.

**Comment by kevindherman** [comment]
@curtismchale The Google Site Kit page under "Analytics Beta" links to a page not found.

![Screen Shot 2023-05-09 at 1 22 17 PM](https://github.com/proudcity/wp-proudcity/assets/6032426/2755217d-bf80-4839-94f2-b2cfd33890b8)

https://whereinwilliamscounty.com/wp-admin/googlesitekit-dashboard

However, the site kit dashboard (listed above pages with G icon) goes to a viewable sitekit view 
https://whereinwilliamscounty.com/wp-admin/admin.php?page=googlesitekit-splash

**Comment by kevindherman** [comment]
@curtismchale the above item only appears to be an issue for support@ and not info@ so probably a permissions thing at this time.

**Comment by lukefretwell** [comment]
Ditto on specific help menu. If we just want the main 'Analytics' tab to go to 'Insights', that's fine with me.

**Comment by curtismchale** [comment]
@kevindherman @lukefretwell the Site Kit menu will **no longer** show for anyone that's not an admin. The way I have to remove it puts the code in the `wp-proud-admin` plugin. So I hide it, then specifically show it if you're signed in as an admin. If I don't show it for admins, we wouldn't be able to set it up.

**Comment by curtismchale** [comment]
@kevindherman now the link works. We'll have to test this still. It's possible that if you haven't accepted that green button page that the link doesn't work. We'll need to see that happen again and I can look at the URL and create another menu that shows up if we need to accept the terms stuff first before we can see analytics.

@kevindherman @lukefretwell the way to test some of the work above is to sign in as support to either `pc.com` or to `whereinwilliamscounty.com`. I **have not** push this all live, but we could. It's really just CSS changes or changing a number from 3 -> 4 so no major functionality.

**Comment by kevindherman** [comment]
@curtismchale that is the case that if you haven't yet accepted the green button the other link doesn't work.
